File: SuperGol/src/main/java/ar/edu/unq/model/DateMatch.java
package ar.edu.unq.model;

import ar.edu.unq.model.position.Position;

import javax.persistence.Entity;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "dateMatch") //No estoy seguro si se usa camelcase
public class DateMatch {

    private List<Match> matches;
    private Integer number;

    public DateMatch(Integer anInteger) {
        number = anInteger;
        matches = new ArrayList<>();
    }

    public void addMatch(Match aMatch) {
        matches.add(aMatch);
        aMatch.setDateMatch(this);
    }

    public void addGoalsOf(Player aPlayer, Position playersPosition, Integer numberOfGoals) {
        new DateMatchGoalCounter(aPlayer, playersPosition, this, numberOfGoals);
    }

    @OneToMany
    public List<Match> getMatches() {
        return matches;
    }
}

File: super_goal/app/models/date_match.rb
class DateMatch < ActiveRecord::Base

  has_many :matches
  belongs_to :fixture

  def add_match(a_match)
    matches << a_match
    a_match.date_match = self
  end

  def has_match?(a_match)
    matches.include? a_match
  end

  def add_goals_of(a_player, a_position, a_number_of_goals)
    GoalsCounter.new(player: a_player, position: a_position, date_match: self, number_of_goals: a_number_of_goals)
  end

  def as_json(options = nil)
    json = super options
    json[:matches] = matches
    json
  end

end